### PR TITLE
Kernel: Try to find a good filename for dumping perfcore

### PR DESCRIPTION
**Pull request by MaxWipfli**
Currently, if the Kernel tries to dump a perfcore to disk and a file with that name already exists, it will just ignore the error, meaning
the performance events will just be lost.

This patch changes the behvaior of `Process::dump_perfcore()` so it appends a dot and a digit after the filename if the "main" filename is already taken. This also adds some more error logging to `dump_perfcore()`.

This annoyed me for too long, and I finally got around to fixing it. :^)

### CODE REVIEW

File: Kernel/Process.cpp [hunk 1]
```diff
@@ -533,21 +533,37 @@ bool Process::dump_perfcore()
...
-    if (description_or_error.is_error())
+    auto base_filename = String::formatted("perfcore.{}", pid().value());
+
+    auto description_or_error = VirtualFileSystem::the().open(base_filename, O_CREAT | O_EXCL, 0400, current_directory(), UidAndGid { uid(), gid() });
+
+    for (size_t attempt = 1; attempt < 10 && description_or_error.is_error(); ++attempt)
+        description_or_error = VirtualFileSystem::the().open(String::formatted("{}.{}", base_filename, attempt), O_CREAT | O_EXCL, 0400, current_directory(), UidAndGid { uid(), gid() });
+
+    if (description_or_error.is_error()) {
+        dbgln("Failed to generate perfcore for pid {}: Could not find non-existing filename.", pid().value());
```
Comment (alimpfard) [inline Kernel/Process.cpp:544]: "Could not find non-existing filename" doesn't seem very nice, maybe just say "could not generate a filename for the perfcore" or something to that effect

File: Kernel/Process.cpp [hunk 2]
```diff
@@ -533,21 +533,37 @@ bool Process::dump_perfcore()
...
-    if (!m_perf_event_buffer->to_json(builder))
+    if (!m_perf_event_buffer->to_json(builder)) {
+        dbgln("Failed to generate perfcore for pid {}: Could not serialize performance events to JSON.", pid().value());
         return false;
+    }
 
     auto json = builder.build();
-    if (!json)
+    if (!json) {
+        dbgln("Failed to generate perfcore for pid {}: Could not not allocate buffer.", pid().value());
```
Comment (alimpfard) [inline Kernel/Process.cpp:557]: Failing to not allocate a buffer is quite an interesting problem :P

**Review by alimpfard — COMMENTED** [review]
The logic looks fine, let's just fix some silly debug prints lol

**Review by alimpfard — COMMENTED** [review]
The logic looks fine, let's just fix some silly debug prints lol

File: Kernel/Process.cpp (hunk 1 above)
Comment (MaxWipfli) [reply Kernel/Process.cpp:544]: Yeah I struggled with that one. Thanks for the suggestion.

File: Kernel/Process.cpp (hunk 2 above)
Comment (MaxWipfli) [reply Kernel/Process.cpp:557]: Yes, indeed.

**Review by alimpfard — APPROVED** [review]

**Pull request merged**